### ISSUE TITLE
Swap out the Initialize method for proper constructors

### ISSUE THREAD

**Issue by christopher-henderson** [zmap/zlint#606]
The `LintInterface` requires the method [`Initialize`](https://github.com/zmap/zlint/blob/7e75dc35e04f682f0d0eb3de6d4af49ccd5db5af/v3/lint/base.go#L27). It seems very reasonable to allow for struct initialization, however the `Initialize` method runs into several issues:

1. This is really just a constructor, and it's a bit awkward to have the constructor be called into post allocation in such a way.
2. `Initialize` returns an `error` during registration, [which immediately halts registration and exits the program with the raw error message returned](https://github.com/zmap/zlint/blob/7e75dc35e04f682f0d0eb3de6d4af49ccd5db5af/v3/cmd/zlint/main.go#L90). This is, in a way, even worse than simply panicking because all stack information is lost - we have no idea which lint produced the error.
3. ...except we sort can figure out which lint caused the error because there are currently a grand total of two lints that actually do anything in their `Initialize` (and these actions are really just normal constructor stuff) - [lint_dnsname_bad_character_in_label](https://github.com/zmap/zlint/blob/7e75dc35e04f682f0d0eb3de6d4af49ccd5db5af/v3/lints/cabf_br/lint_dnsname_bad_character_in_label.go#L40) and [lint_rsa_public_exponent_not_in_range](https://github.com/zmap/zlint/blob/7e75dc35e04f682f0d0eb3de6d4af49ccd5db5af/v3/lints/cabf_br/lint_rsa_public_exponent_not_in_range.go#L46).

### But the biggest problem is
4. Due to our use of a global registry all lints are singleton objects at runtime. This is fine at this exact moment in time, however it has caused extreme pain when attempting to implement #592. In #592 I like to do something like the following:

```go
type Configurable interface {
    Configure() interface{}
}
```

...so that a lint can receive further context as necessary. It simply gives us a struct to deserialize a configuration format into.

```go
struct MyLint {
    configuration struct {
        the_rum_is_gone bool
    }
}

func (m *MyLint) Configure() interface{} {
    return &m.configuration
}

func (m *MyLint) Execut(c *x509.Certificate) *lint.Result {
    if m.configuration.the_rum_is_gone {
        return &lint.Result{Status: lint.Err, Details: "Why is the rum gone!?"}
    }
    return &lint.Result{Status: lint.Pass}
}
```

However, this simply does not work for well for unit and integration testing. It becomes a huge race condition to your tests in the context of a particular configuration because all of our lints are global, mutual, singletons.

## Proposal

1. Remove `Initialize` from the `LintInterface`

2. Change the `Lint` field in the `Lint` struct to be a constructor that can return new instances as is necessary

```go
// A Lint struct represents a single lint, e.g.
// "e_basic_constraints_not_critical". It contains an implementation of LintInterface.
type Lint struct {

        ...
        // A constructor which returns the implementation of the lint logic.
	Lint func() LintInterface `json:"-"`
}
```

I believe that this constructor should be infallible (I.E. does not return an error). It does not seem like we would actively encourage any constructors that could reasonably fail (that is, and error is reasonable if it were reading from its own lint specific files or reaching out to the network, but we just don't seem to have the use case to do that in an adhoc per-lint manner). For example, in the one lint that currently could ever produce an error it would be perfectly reasonable (desirable, even) to panic because it would be failing to [construct a static regular expression](https://github.com/zmap/zlint/blob/7e75dc35e04f682f0d0eb3de6d4af49ccd5db5af/v3/lints/cabf_br/lint_dnsname_bad_character_in_label.go#L40).
